assets/javascripts/discourse/initializers/custom-wizard-redirect.js.es6

```diff
@@ -15,30 +19,26 @@ export default {
       window.location.href = wizardUrl;
     });
 
-    const ApplicationRoute = requirejs("discourse/routes/application").default;
+    withPluginApi("0.8.36", (api) => {
+      api.onAppEvent("page:changed", (data) => {
+        const currentUser = container.lookup("service:current-user");
 
-    ApplicationRoute.reopen({
-      actions: {
-        willTransition(transition) {
-          const redirectToWizard = this.get("currentUser.redirect_to_wizard");
-          const excludedPaths = this.siteSettings.wizard_redirect_exclude_paths
+        if (currentUser) {
+          const redirectToWizard = currentUser.redirect_to_wizard;
+          const excludedPaths = siteSettings.wizard_redirect_exclude_paths
             .split("|")
             .concat(["loading"]);
-
           if (
             redirectToWizard &&
-            (!transition.intent.name ||
-              !excludedPaths.find((p) => {
-                return transition.intent.name.indexOf(p) > -1;
-              }))
+            data.currentRouteName !== "customWizardStep" &&
```

Comment:
@merefield Why are we adding in a customWizardStep check here?

Reply:
Not sure what planet I was on when I wrote that as no precedent I can see in prior code, but it's to prevent an attempt to redirect when you are currently inside a wizard.

Reply:
@angusmcleod now I remember.  It's to prevent an infinite loop.  ie once you arrive at the wizard, the redirect stops.

Reply:
I've put it back.